### PR TITLE
Use right lane invalids for spine

### PR DESCRIPTION
**Pull request by jallen-NextDroid**
Now the closest left and right points are used to determine spine validity. If either is invalid, the spine will be invalid

### CODE REVIEW

**Review by arjunskND — APPROVED** [review]

File: demo/laneLoader.js [hunk 1]
```diff
@@ -249,14 +249,19 @@ async function createLaneGeometries (lanes, supplierNum, annotationMode, volumes
...
         const closestLeftIndex = geometryLeft.vertices.reduce((acc, current, i) => {
           return current.distanceTo(spinePoint) < geometryLeft.vertices[acc].distanceTo(spinePoint) ? i : acc;
         }, 0);
 
-        spineValidities.push(leftValidities[closestLeftIndex])
+        const closestRightIndex = geometryRight.vertices.reduce((acc, current, i) => {
+          return current.distanceTo(spinePoint) < geometryRight.vertices[acc].distanceTo(spinePoint) ? i : acc;
+        }, 0);
+
+        spineValidities.push(leftValidities[closestLeftIndex] || rightValidities[closestRightIndex]);
```
Comment (arjunskND) [inline demo/laneLoader.js:264]: Shouldn't it be `leftValidities[closestLeftIndex] || rightValidities[closestRightIndex` ?

**Review by arjunskND — CHANGES_REQUESTED** [review]
Not sure if you have negated leftValidities and rightValidities, ie. does true mean `Valid` ? If so shouldn't it be `left && rightValid`, meaning both should be valid for spine to be valid?

**Comment by jallen-NextDroid** [comment]
It's 0 for valid, 1 for invalid. Kinda backwards I suppose, maybe I'll change the variable name to be more clear

**Review by bcasselsND — APPROVED** [review]

**Comment by bcasselsND** [comment]
Yuck. You could rename to "invalid" if you need 0 to be valid and 1 invalid.  But why not switch to 0 invalid and 1 valid?

Otherwise you end up with double negatives `!invalid`.

**Review by arjunskND — APPROVED** [review]

**Pull request merged**